codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoDelegator.java

```diff
@@ -93,16 +93,39 @@ public void useShapeTestWriter(Shape shape, Consumer<GoWriter> writerConsumer) {
         filename = b.toString();
 
         symbol = symbol.toBuilder()
-            .definitionFile(filename)
-        .build();
+                .definitionFile(filename)
+                .build();
+
+        useShapeWriter(symbol, writerConsumer);
+    }
+
+    /**
+     * Gets a previously created writer or creates a new one for the a Go public package test file for the associated
+     * shape.
+     *
+     * @param shape          Shape to create the writer for.
+     * @param writerConsumer Consumer that accepts and works with the file.
+     */
+    public void useShapePublicTestWriter(Shape shape, Consumer<GoWriter> writerConsumer) {
+        Symbol symbol = symbolProvider.toSymbol(shape);
+        String filename = symbol.getDefinitionFile();
+
+        StringBuilder b = new StringBuilder(filename);
+        b.insert(filename.lastIndexOf(".go"), "_public_test");
+        filename = b.toString();
+
+        symbol = symbol.toBuilder()
+                .definitionFile(filename)
+                .namespace(symbol.getNamespace() + "_test", symbol.getNamespaceDelimiter())
+                .build();
 
         useShapeWriter(symbol, writerConsumer);
```

Comment:
minor semantics, Should `Public` be `Exported` since that is the phrasing in Go?